alphagov/paas-rubbernecker
# [#164248858] Put scheduled sticker on stories based on `scheduled:<date>` story label

## What

This PR adds a new sticker called scheduled which can show a calendar icon with a date next to it. It means the story should be released on or after the given date.

If the blocker message contains any of the following expressions, then a
scheduled sticker will be attached to the story:
  - until 2/9
  - until 02/09
  - until 02/09/2019
  - until 2019/09/02
  - until 2019-09-02

Other prepositions are also supported: "on or after", "after, "before", "on"

If "before" is used then the scheduled day will be the next day.

We will stop at the first successfully parsed message, so any additional dates are ignored.

We will only add a scheduled sticker if the scheduled date is in the future.

The blocker sticker won't be shown if there are no other blockers than one containing a scheduled date.

![screenshot 2019-02-26 at 16 00 37](https://user-images.githubusercontent.com/868104/53426911-ccebb380-39df-11e9-92ec-b198a55e0d1e.png)

Also I fixed the startup process, so we don't have to wait 20 seconds for the stories to be loaded.

Also a priority field has been introduced on the stickers, so we can show the scheduled one always on the right.

## How to review

1. Run the Rubbernecker app locally.
    ```
    export PIVOTAL_TRACKER_PROJECT_ID=1275640
    export PIVOTAL_TRACKER_API_TOKEN=xxx
    export PAGERDUTY_AUTHTOKEN=xxx

    make build
    ./bin/rubbernecker
    ```

1. Add e.g. "Blocked until 1/4" as a blocker to any story in play and check whether the sticker is properly added.

## Who can review

Not me.
